src/modules/LanUdevAuto.rb

```diff
@@ -194,39 +198,43 @@ def Import(settings)
     end
 
     def Write
-      comment = [
-        "# Generated by autoyast",
-        "# program run by the persistent-net-generator.rules rules file.",
-        "#",
-        "# You can modify it, as long as you keep each rule on a single line."
-      ]
+      template = "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", %s==\"%s\", NAME=\"%s\""
+
       rules = []
-      Builtins.foreach(@udev_rules) do |rule|
-        rules = Builtins.add(
-          rules,
-          Builtins.sformat(
-            "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", %1==\"%2\", NAME=\"%3\"",
-            Ops.get_string(rule, "rule", ""),
-            Ops.get_string(rule, "value", ""),
-            Ops.get_string(rule, "name", "")
-          )
-        )
+
+      @udev_rules.each do |rule|
+        opt = rule["rule"] || ""
+        value = rule["value"] || ""
+        devname = rule["name"] || ""
+
+        rules << template % [opt, value.downcase, devname]
       end
-      if Ops.greater_than(Builtins.size(rules), 0)
-        Builtins.y2milestone(
-          "Writing AY udev rules for network (will replace original rules from 1st stage)"
-        )
-        if AllowUdevModify()
-          SCR.Write(path(".udev_persistent.rules_comment"), comment)
-          SCR.Write(path(".udev_persistent.rules"), rules)
-        end
+
+      if !rules.empty? && AllowUdevModify()
+        SetAllLinksDown()
+
+        log.info("Writing AY udev rules for network")
+
+        comment = [
+          "# Generated by autoyast",
+          "# program run by the persistent-net-generator.rules rules file.",
+          "#",
+          "# You can modify it, as long as you keep each rule on a single line."
+        ]
+
+        SCR.Write(path(".udev_persistent.rules_comment"), comment)
+
+        write_update_udevd(rules)
+
+        SCR.Execute(path(".target.bash"), "udevadm settle")
       else
-        Builtins.y2milestone(
-          "No udev rules defined by AY, nothing to write (keep original file)"
-        )
+        log.info("No udev rules defined by AY")
```

Comment:
Then it wasn't true even before. ```AllowUdevModify``` is not newly added. 

I'll rephrase the message.